src/installer.ts

```diff
@@ -53,18 +190,23 @@ export async function getNode(
     versionSpec = resolveLtsAliasFromManifest(versionSpec, stable, manifest);
   }
 
-  if (isLatestSyntax(versionSpec)) {
+  // TODO: 183-189 and 193-194   seems to be the same. Why do we need them?
+  if (isLatestSyntax(versionSpec) || distribution == Distributions.CANARY) {
     nodeVersions = await getVersionsFromDist(versionSpec);
     versionSpec = await queryDistForMatch(versionSpec, arch, nodeVersions);
-    core.info(`getting latest node version...`);
+    core.info(
+      `getting ${
+        distribution == Distributions.CANARY ? 'v8-canary' : 'latest'
+      } node version ${versionSpec}...`
+    );
   }
 
-  if (isNightly && checkLatest) {
+  if (distribution === Distributions.NIGHTLY && checkLatest) {
     nodeVersions = await getVersionsFromDist(versionSpec);
     versionSpec = await queryDistForMatch(versionSpec, arch, nodeVersions);
   }
 
-  if (checkLatest && !isNightly) {
+  if (checkLatest && distribution !== Distributions.NIGHTLY) {
```

Comment:
I think we can change it to `if (checkLatest && distribution === Distributions.DEFAULT)`.